### ISSUE TITLE
TypeError when getting slurm scheduler info

### ISSUE THREAD

**Issue by rdocking** [bcbio/bcbio-nextgen#2742]
Running bcbio 1.1.4 on a slurm scheduler:

```
[2019-04-03T18:40Z] n105: Couldn't get machine information from resource query function for queue 'all' on scheduler "slurm"; submitting job to queue
Traceback (most recent call last):
  File "/projects/rdocking_prj/software/bcbio-nextgen/data/anaconda/lib/python3.6/site-packages/bcbio/provenance/system.py", line 48, in _get_machine_info
    return sched_info_dict[parallel["scheduler"].lower()](parallel.get("queue", ""))
  File "/projects/rdocking_prj/software/bcbio-nextgen/data/anaconda/lib/python3.6/site-packages/bcbio/provenance/system.py", line 67, in _slurm_info
    mem = float(mem.replace('+', ''))
TypeError: a bytes-like object is required, not 'str'
```

Seems to be a Python3 issue - 

https://stackoverflow.com/questions/33054527/typeerror-a-bytes-like-object-is-required-not-str-when-writing-to-a-file-in

Note that this didn't crash my run - thought I'd flag it though. Thanks!

**Comment by chapmanb** [comment]
Rod;
Thanks much for catching this. We've pushed a fix for this along with #2741 so hope that you'll now get clean error free runs on python 3. Please let us know if you find anything else at all and thanks again.

**Comment by rdocking** [comment]
Confirmed that this is working in the latest development version - thanks for the quick fix!
